amundsen_application/static/js/components/SearchPage/tests/index.spec.tsx

```diff
@@ -273,13 +270,47 @@ describe('SearchPage', () => {
       wrapper.instance().componentDidUpdate(mockPrevProps);
     });
 
-    it('calls setState', () => {
-      expect(setStateSpy).toHaveBeenCalledWith({ selectedTab: ResourceType.table });
-    });
+    describe('called with a new search term', () => {
+      it('calls setState', () => {
```

Comment:
We call `setState` regardless of what happens with the search term so it's misleading to bundle that into this case. These two checks can be separate tests and don't have to be bundled into the same block:
```
it('calls setState', () => {
    expect(setStateSpy).toHaveBeenCalledWith({ selectedTab: ResourceType.table });
});

it('calls searchAll if called with a new search term', () => {
   expect(searchAllSpy).toHaveBeenCalledWith(mockSanitizedUrlParams.term, mockSearchOptions);
});
```
 
However one thing we can keep is the fact that you added some more explicit titles for the tests, to help clarify the conditional logic that we're testing.